\documentclass[12pt, letterpaper]{article}
\usepackage[utf8]{inputenc}
\usepackage[margin=1in]{geometry}
\usepackage[super]{nth}

\usepackage{lineno}
\usepackage[
singlelinecheck=false
]{caption}
\usepackage{amsmath}
\usepackage{amsfonts}
\usepackage{bm}
\usepackage{bbm}
\usepackage{graphicx}
\usepackage{csvsimple}
\usepackage[section]{placeins}
\usepackage{lineno}
\usepackage{natbib}
\usepackage[T1]{fontenc}
\usepackage[breaklinks]{hyperref}
\usepackage{microtype}

\title{KIN: A method to infer relatedness from low-coverage ancient DNA}
\author{Divyaratan Popli, Stéphane Peyrégne, Benjamin M. Peter}
%\date{5 August 2021}
\linenumbers

\setlength{\parskip}{1em}
\setlength{\parindent}{0em}

\newcommand{\BZ}{\mathbf{Z}}
\newcommand{\BD}{\mathbf{D}}
\newcommand{\BN}{\mathbf{N}}
\newcommand{\BH}{\mathbf{H}}
\newcommand{\Btheta}{\pmb{\theta}}


\hbadness=99999
\begin{document}

\maketitle


\begin{abstract}

\noindent Genetic kinship of ancient individuals can provide insights into their culture and social
hierarchy, and is relevant for downstream genetic analyses. However, estimating
relatedness from ancient DNA is difficult due to low-coverage, ascertainment bias, or
contamination from various sources. Here, we present KIN, a method to estimate the
relatedness of a pair of individuals from the identical-by-descent segments they share.
KIN accurately classifies up to 3rd-degree relatives using at least 0.05x sequence coverage
and differentiates siblings from parent-child. It incorporates additional models to adjust
for contamination and detect inbreeding, which improves classification accuracy.
\end{abstract}

\section{Introduction}

\subsection{Why study relatedness?}

Identifying related individuals is a common task in genetic studies. Relatedness is of direct interest in e.g. DNA forensics, where familial search can aid in solving criminal cases, and to identify unknown deceased persons \cite{murphy_law_2018,ram_genealogy_2018}. Genetic paternity tests have an important application in resolving family relation, e.g. in establishing relationship between an individual applying for immigration and the claimed relatives \cite{egeland_beyond_2000}. It is also an essential step in population genetics and association studies, where samples are typically assumed to be independent random draws from the population. For animal and plant breeders and conservation biologists, reconstructing pedigrees and finding related individuals is important to avoid inbreeding and ensure diversity \cite{habier_impact_2007,oliehoek_estimating_2006,kardos_measuring_2015}.

In ancient DNA studies, relatedness can be used to identify bones and teeth belonging to the same individual. Given adequate familiarity with the subject, relatedness can provide an understanding of an ancient society's social structures, mobility and inheritance rules ~\cite{baca_ancient_2012,mittnik_kinship-based_2019,sikora_ancient_2017}.


\subsection{Approaches to estimate relatedness from high-coverage data}

Commonly, pairs of related individuals are identified by looking for parts of the genome that are identical by descent (IBD), i.e. inherited from a recent common ancestor. Due to the laws of Mendelian segregation, each parent will share exactly one set of chromosomes IBD with their offspring, while subsequent recombination means that a grandparent will, on average share a quarter of their genome with a grand-child. Along the genomes of a pair of diploid individuals, there are three IBD states possible at any given position: the individuals share zero, one or two chromosomes IBD. The genome-wide proportions of these states (usually referred to as $k_0$, $k_1$, $k_2$, so that $k_0+k_1+k_2=1$) can be used to infer the degree and nature of relatedness for a pair of individuals. For example, a pair of siblings are expected to have all three possible IBD states with proportions of 0.25,0.5,0.25, respectively (Fig. \ref{fig0:schematic}). These IBD probabilities can directly be used to categorize their relatedness as shown in Table \ref{tab:Table 1}. One can also use these probabilities to estimate the coefficient of relatedness $r$, which is defined as the proportion of the genome that is IBD. In the absence of inbreeding, this would be calculated as $r= k_1/2 + k_2$.

However, since it is not possible to directly observe IBD segments, a common approach is to first identify segments of the genome that are Identical by State (IBS) and to use population allele frequencies obtained from an out-of-sample reference panel to calculate the probability of IBD given IBS. There are several methods that incorporate reference panel allele frequencies, phase information, recombination maps, or genotype calls to co-estimate IBD and the relatedness coefficient \cite{huff_maximum-likelihood_2011,li_relationship_2014,thornton_estimating_2012, boehnke_accurate_1997,lynch_estimation_1999, purcell_plink_2007,manichaikul_robust_2010,gusev_whole_2009,nyerki_optimized_2022,browning_fast_2011,li_accurate_2014}.

\begin{figure}[ht!]
    \includegraphics[width=16.5cm]{plots/inkscape_finalImg/schematic_sib_rev.png}
    \centering
    \caption{IBD sharing between siblings without and with runs of homozygosity (ROH). (A) Schematic of chromosomes for two parents (Par1, Par2). (B) Schematics of recombinant chromosomes of two children (Sib1, Sib2). (C)  Expected differences between Sib1 and Sib2 ($p$) along the chromosome. In both cases, $p$ can take values of $p_0$, $p_1$ or $p_2$, which are the expected proportion of differences in IBD states 0,1 and 2, respectively.  (D-F): Same as A-C, except parent 1 is assumed to be homozygous.}
    \label{fig0:schematic}
\end{figure}


\begin{table}[ht!]
\caption{\label{tab:Table 1}IBD sharing probabilities for different relations in absence of inbreeding}
\begin{tabular}{|c|c|c|c|}
    \hline
    Relatedness & $k_0$ & $k_1$ & $k_2$\\
    \hline
    Unrelated & 1 & 0 & 0\\
    \hline
    3rd Degree & 0.75 & 0.25 & 0\\
    \hline
    2nd Degree & 0.5 & 0.5 & 0\\
    \hline
    Siblings & 0.25 & 0.5 & 0.25\\
    \hline
    Parent-Child & 0 & 1 & 0\\
    \hline
    Identical/twins & 0 & 0 & 1\\
    \hline
\end{tabular}
\label{table1}
\end{table}

\subsection{Approaches that address problems with ancient DNA data}
One major issue with applying the above-mentioned methods to ancient DNA data is that the sequence coverage is typically low, making it difficult to obtain accurate genotype calls.
Several methods surmount this problem by using genotype likelihoods \cite{lipatov_maximum_2015,korneliussen_ngsrelate_2015}. In this way, it is possible to account for the uncertainty in genotype calls by summing over all possible genotypes, weighted by their genotype likelihoods. However, these approaches typically still require at least 2x coverage, since genotype likelihoods may be imprecise at lower coverages \cite{korneliussen_angsd_2014} . Ancient DNA analyses often face additional challenges such as the unavailability of  reference panels to estimate population allele frequencies, contamination with present-day DNA \cite{peyregne_present-day_2020-1}, and an ascertainment bias caused by  DNA capture approaches \cite{prufer_computational_2010, vai_kinship_2020}.

Several methods have been proposed to estimate relatedness without a reference panel from ancient DNA, but they  require either $>4$x coverage \cite{waples_allele_2019}, or a large sample size to get an estimate of allele frequencies in the population from which the sampled individuals originate \cite{theunert_joint_2017}. A second issue is contamination. If the contamination stems from another population, contaminated data will look more dissimilar to other individuals from the analyzed population, and hence relatedness will be underestimated. In addition, some analyzed  genomes may have long runs of homozygosity (ROH), for example due to a small population size, or recent inbreeding. Long ROH cause related individuals to seem genetically more similar to each other, but do not affect the genetic distance between unrelated individuals.

Moreover, ancient DNA is commonly captured with a SNP array that enriches for informative variants. Particularly, methods based on the fraction of sites in different IBS states are sensitive to the ascertainment bias caused by this non-random selection of targeted sites \cite{waples_allele_2019}.

READ \cite{kuhn_estimating_2018} addresses several of the issues encountered in the analysis of ancient DNA. In particular, the lack of genotype calls is dealt with by randomly sampling alleles from each individual. A string of these alleles at each position (called pseudo-haploids) are then compared to other individuals to estimate  average pairwise genetic distances, which in turn are used to infer relatedness. However, READ only infers the degree of relatedness, and only up to second degree.

\subsection{How our method works}
Here, we present KIN (Kinship Inference), a Hidden Markov Model (HMM)-based approach to estimate genetic kinship and IBD from low-coverage ancient DNA data. KIN can detect up to \nth{3}-degree relatives, and differentiates between siblings and parent-child relationships. KIN is also able to take into account ROH and contamination, and is not sensitive to SNP ascertainment. We validate the performance of KIN using simulations and show that we are able to infer relatedness in real data from two datasets: a group of Neandertals and a group of Bronze age individuals.


\section{Algorithm}\label{new_approaches}
To infer relatedness, KIN fits one HMM for each pair of individuals and for each possible relatedness. The KIN-HMM infers IBD sharing between a pair of low-coverage individuals, optionally taking ROH tracts and contamination estimates in each individual into account. The best-fit model is then assigned as the inferred relatedness. If the locations of ROH tracts are unknown, we provide another HMM (ROH-HMM) to coarsely estimate the location of ROH for samples with sufficient coverage ($\geq 0.1x$). Our method is available on \url{https://github.com/DivyaratanPopli/Kinship_Inference} along with a python package (KINgaroo) to generate the input files for the models directly from bam files.


\subsection{Model description}\label{method_overview}
The goal of KIN-HMM is to infer how two individuals are related via the patterns of shared IBD states along a pair of genomes. For this purpose,  we subdivide the genomes of a pair of individuals into $L$ large genomic windows (typically of size 10Mb), and infer the pattern of IBD-sharing for each relatedness case we consider (Unrelated, $5^{th}$ Degree, $4^{th}$ Degree, $3^{rd}$ Degree, Grandparent-Grandchild, Avuncular, Half-siblings, Parent-Child, Siblings, and Identical). We then compare the likelihood between all models, and classify each pair to the model with the highest likelihood. We also return the most likely locations of IBD tracts (using standard Viterbi algorithm), and the IBD state posterior probabilities.

The details of the likelihood computation are given in section \ref{ll}. As is the case for any HMM, KIN-HMM requires a set of emission (section \ref{B}) and transition probability matrices. We assume the transition matrix for each relatedness case is fixed. Our emissions include a vector of parameters ($\delta$) that describe the variance in the data for each IBD state that we infer from the data (section \ref{B}).

\subsection{Input of KIN-HMM}\label{hmm_input}
The inputs of our algorithm are i) the number of overlapping sites for the $w$-th window $N_w$ for which both samples have at least one read available, ii) the number of pairwise differences $D_w$ at these sites, and iii) the probability of ROH in windows, by default obtained from ROH-HMM described in section \ref{roh}.

For high-coverage data, $D_w$ can be directly obtained by comparing genotypes, but for low-coverage samples, $D_w$ needs to be estimated from the sequencing data. The simplest approach is to randomly-sample a read from each position \cite{haak_massive_2015,prufer_complete_2014,green_draft_2010}. However, such an approach may result in loss of data, and hence we estimate $D_w$ by implicitly summing over all possible samplings:
\begin{align}\label{eq:x}
D_w &= \sum_{s=1}^{N_w} \nu_i(s) (1-\nu_j(s)) + (1-\nu_i(s)) \nu_j(s)
\end{align}
Here, $\nu_i(s)$ and $\nu_j(s)$ are the proportions of reads carrying the derived allele at SNP index $s$ for individuals i and j respectively.
Throughout, we will use bold-face notation to refer to the vector (or matrix) collecting  all the terms, e.g. $\BD = (D_1, D_2, \dots D_L)$.


\subsection{Log likelihood of the KIN-HMM}\label{ll}
The KIN-HMM uses $\BD$ and $\BN$ to classify each window into three hidden states $Z_w \in$ ($0$, $1$, $2$), reflecting zero, one or two shared chromosomes IBD, respectively. To take ROH into account, we define the variable $H_w \in$ ($0$, $1$, $2$) that designates that zero, one or both individuals are homozygous in window $w$. Since $H_w$ is unobserved, in practice we use the estimates from ROH-HMM: $h_{wj} = P(H_w = j)$.

There are three additional model parameters, $\bm{\pi}, \mathbf{A}$ and $\bm{\delta}$. The transition matrix $\mathbf{A}$ gives the probability of moving from state $i$ to state $j$, given by $a_{ij}$, and is fixed and estimated from simulations for each relatedness case (section \ref{simulat}). The initial probabilities $\bm{\pi}$ give the probabilities of being in each state $Z_0$ (at the beginning of each chromosome), which we set to the stationary distribution of the transition matrix for simplicity. The overdispersion-parameter $\bm{\delta}$ takes into account that SNPs in each window vary in their allele frequency (see next setion).  For compactness of notation, we group the fixed parameters: $\theta = (\BN, \mathbf{A}, \bm{\pi})$.

Thus, the complete data likelihood for the HMM is

\begin{align}\label{eq:1}
\log P(\BD,\BZ|\BH, \theta, \bm{\delta}) &= \log P(\BD|\BZ,\BH, \theta, \bm{\delta}) + \log P(\BZ |\theta) \nonumber\\
&= \sum_w \log P(D_w|Z_w,H_w, N_w, \bm{\delta}) + \sum_w \log P(Z_w |Z_{w-1},\bm{A}) + \log P(Z_0|\bm{\pi})\text{.}
\end{align}

Here, $\BZ$ is not dependent on $\BH$ and $\bm{\delta}$.

\subsection{Emission probability}\label{B}

Using this setup, we can isolate the emissions $P(D_w | Z_w, H_w ,\theta, \bm{\delta})$ from equation \ref{eq:1} and optimize them for $\bm{\delta}$. The simplest model is to assume that sites in each window are equally distributed and independent, in this case we could use the  binomial likelihood:
$$P(D_w|Z_w, H_w, N_w) \sim \text{Binom}[D_w ; p(Z_w, H_w), N_w] \text{,}$$

where, $p$ is the proportion of differences expected for a particular IBD and ROH state. If the two individuals are unrelated in a particular window (i.e. $Z_w = 0$), then the expected proportion of pairwise differences depends solely on the population history, and we denote this proportion with $p_0$. If the two individuals share one or even both copies of the genome IBD, we would expect the proportion of differences to be reduced to $p_1 = \frac{3}{4} p_0$, and $p_2 = \frac{1}2 p_0$, respectively, since either one or two of the four possible comparisons will be between identical chromosomes \cite{kuhn_estimating_2018}. Thus, $p(Z_w=i, H_w=0) = p_i$.

The proportion of differences between unrelated individuals $p_0$ is an important parameter. We follow READ \cite{kuhn_estimating_2018} and estimate $p_0$ as the median of differences for all possible pairs of individuals, which works well if the majority of individuals in the sample are unrelated.

The presence of long tracts of homozygosity resulting from recent inbreeding adds an additional complication, as the number of shared chromosomes may be overestimated \cite{weir_genetic_2006}. For example, when considering two bones from the same individual, we would expect the entire genome to have a pairwise difference of $p_2$, because two out of the four compared chromosomes are identical copies. However, in inbred regions all four chromosomes will be identical, and so the expected pairwise differences are zero ($p_4$ in equation \ref{eq:2}). Note however, that $p_0$ does not depend on the presence of ROH, since all comparisons are between unrelated chromosomes even if both individuals are homozygous at a particular locus.

Taken together, we can summarize $\bm{p}$ in the following matrix, where rows give the state of $Z_w$, and columns of $H_w$:
\begin{equation}\label{eq:2}
    p(Z_w, H_w) = \left[\begin{array}
{rrr}
p_0 & p_0 & p_0 \\
p_1 & p_2 & p_4 \\
p_2 & p_4 & p_4
\end{array}\right]
\end{equation}

As explained above, we would expect $p_4$ to be zero. However, as we do all our calculations in large windows, the start/end positions of windows may not coincide with that of ROH tracts, and we found that we obtain better results by setting $p_4 = \frac{p_2}{2}$, to take into account that many windows will only partially have four comparisons between identical chromosomes.

The effect of these considerations is that even though we have nine possible combinations of $Z_w$ and $H_w$ for each window, there are actually only four unique $p$-parameters $p_i$ with $i$ $\in$ (0,1,2,4).

\paragraph{Beta Binomial Model:}
We empirically find that the data often has considerably higher variance than would be expected from a binomial model (Fig. S2). We take this into account by adding an over-dispersion parameter $\bm{\delta}$. Just like $p(Z_w,H_w)$, $\delta(Z_w,H_w)$ depends on the number of chromosomes compared, and so each of the  four $p_i$ has a corresponding $\delta_i$ parameter.

Taken together, our emission probabilities are
\begin{align}\label{eq:3}
P(D_{w}|Z_w,H_w,N_w, \bm{\delta}) &\sim BB[D_w; p_i, \delta_i, N_w] \nonumber\\
&= \binom{N_w}{D_w}\frac{B(D_w+p_i \delta_{i}, N_w-D_w+ \delta_{i}(1-p_{i}))}{ B(p_{i}\delta_{i}, (1-p_{i})\delta_{i})},
\end{align}
where $i$ is fully determined by the combination of $Z_w$ and $H_w$ (see equation \ref{eq:2}).

This parameterization of the beta distribution in terms of expected value $p$ and over-dispersion $\delta$ is also called the Balding-Nichols-model \cite{balding_method_nodate}, and is distinct from the more common parameterization in terms of $\alpha$ and $\beta$. We use this equation even if preprocessing steps (see equation \ref{eq:x} and section \ref{contam}) result in non-integer $D_w$ and $N_w$, in which case we approximate the binomial coefficient using Gamma functions.

\subsection{Estimation of \texorpdfstring{$\delta$}{}}\label{delta}
We estimate the $\delta$-parameters using an Expectation-Maximization (EM) algorithm \cite{dempster_maximum_1977}.

\paragraph{Initialization:}
The value of $\delta_i$ is unknown to start with, and we set it to a random value between 0 and 1000.

\paragraph{Expectation step:}
In the $t$-th iteration, we calculate the posterior probability of each IBD state in each window $\gamma^{(t)}_{wi} = P(Z_w=i | D_w, H_w, \theta_i, \delta_i^{(t)})$ using the forward-backward algorithm, where $\delta_i^{(t)}$ is the current estimate for $\delta$ for a given IBD state.

\paragraph{Maximization step:}
The only free parameters we estimate in the maximization step are the over-dispersion parameters $\delta_i$. We do this optimization using a cost function, which is the log-emission probability weighted by the posterior probabilities of the hidden states $\gamma_{wj}$ and optionally the ROH state-probabilities $h_{w\omega}$ obtained from the ROH-HMM.


\begin{align}\label{eq:4}
\mathcal{C} &= \mathbb{E}[\log P(D_w|Z_w, H_w, \theta, \delta^{(t-1)})]\nonumber\\
&= \sum_{w=1}^L \sum_{j=0}^2\sum^2_{\omega=0} \log P(D_{w}|N_w, \delta(j ,\omega)^{(t-1)}, p(j,\omega))h_{w\omega}\gamma_{wj}
\end{align}


Using equation \ref{eq:2}, we simplify this by grouping all the terms that have the same number of pairwise comparisons between identical chromosomes, which would result in the same $p_i$ and $\delta_i$, i.e.

\begin{align}\label{eq:5}
g_{w0} &= \gamma_{w0}\nonumber\\
g_{w1} &= \gamma_{w1} h_{w0}\nonumber\\
g_{w2} &= \gamma_{w2} h_{w0} + \gamma_{w1} h_{w1}\nonumber\\
g_{w3} &= 0 \nonumber\\
g_{w4} &= \gamma_{w2} h_{w1} + \gamma_{w2} h_{w2} + \gamma_{w1} h_{w2},
\end{align}

where $g_{w3}$ is always 0 because there is no case that leads to only three comparisons between identical chromosomes.

So we can rewrite equation \ref{eq:4} as:
\begin{align}\label{eq:6}
\mathcal{C} &= \sum_{i=0}^4 \sum_{w=1}^L \log P(D_{w}|N_w, \delta_i, p_i)g_{wi}
\end{align}

The cost function (equation \ref{eq:6}) has one independent term for each $i$ and so we can separate them and estimate each $\delta_i$ independently using the minimize\_scalar algorithm implemented in \texttt{scipy.optimize} \cite{virtanen_scipy_2020}.

%\paragraph{Constraints on $\delta$ optimization}\label{delta_constraint}

We constrain the optimization space of the $\delta_i$, as unconstrained optimization could result in some confounding of cases. We know that different cases of relatedness have different numbers of IBD states possible. For example, siblings may have all three IBD states present while a Parent-Child has only $Z_w=1$. However, the parent-child model could fit data generated under the sibling model by assigning it a very high $\delta_1$, which would reduce performance (for example, see Fig. S5, S6). We avoid this problem by constraining the $\bm{\delta}$ such that the beta distributions for the different relatedness cases overlap by at most one standard deviation.

\subsection{Model comparison}\label{model_comp}
To infer the most likely relatedness case, we run our model on all relatedness cases mentioned in section \ref{method_overview}, and compare the resulting likelihoods. We output the relatedness corresponding to the maximum likelihood model. Since we compare models where the  parameters are not subset of each other, standard likelihood-ratio theory for nested models cannot be used to obtain confidence intervals. Instead we use the log likelihood ratio between the two best models as a statistic to assess the confidence in our classifications, and use simulations to obtain critical values (Fig. S8).

\paragraph{Grouping of cases:}
Particularly for low-quality data, we may not be able to distinguish all cases. Thus, we group $4^{th}$ and $5^{th}$ degree relatives together with unrelated. Similarly, we group  half-siblings, avuncular and grandparent-grandchild to  $2^{nd}$-degree relatives in the final results. We report the final pairwise classification in the following categories: Unrelated, Third Degree, Second Degree, Parent-Child, Siblings, Identical individuals.


\paragraph{Critical Values:}

To investigate the limits of our method, we plotted the true-positive and false-positive rates for classification of different relatedness in control simulations (without contamination, ROH and ascertainment bias, see section \ref{simulat}) when we use a particular difference in log-likelihood as a cutoff (Fig. S8). The figure shows that for all relatedness cases except for $3^{rd}$-degree, the false positive rate is below $5\%$ even when simply selecting the model with the highest likelihood. We observe that for $3^{rd}$-degree relatives, using a cutoff of 1.0 brings down the false positive rate close to $5\%$ for all coverages except 0.05x. Thus, we recommended using a cutoff of 1 for all cases where ROH and contamination are not a concern.


\paragraph{Example case of siblings:}
In Fig.\ref{fig1:ibd} we show the inferred IBD fragments from different KIN-HMMs when they are applied to simulated data from a pair of siblings. The models for identical, parent-child or unrelated relationships allow for just one IBD state, resulting in a flat line and low likelihood for this data. The other three cases all allow for different IBD-states, but the siblings-model predictions match true IBD states the most, as reflected by the highest log-likelihood and the close correspondence of the inferred and true IBD states.

\begin{figure}[ht!]
    \includegraphics[width=16.5cm]{plots/inkscape_finalImg/IBDplot_title.png}
    \centering
    \caption{Comparison of pairwise difference data and inferred IBD fragments. The top panel shows the proportion of differences in each window along the genome for a pair of simulated siblings. Dashed lines represent $p_0$, $p_1$ and $p_2$ estimates. The second panel shows the true IBD state for each window. The remaining panels show the IBD states predicted by particular relatedness models. The log-likelihood value for each model is shown on upper left corner of the panel. Light and shaded backgrounds represent distinct chromosomes.}
    \label{fig1:ibd}
\end{figure}

\subsection{ROH estimation model}\label{roh}
Our HMM to detect ROH tracts works similarly to the KIN-HMM described above, but in this case we only consider one individual at a time, and only consider positions covered by at least two reads. For each site, we calculate the proportion of reads that carry different alleles, and sum them up in windows along the genome. We call the vector with the number of differences $\mathbf{\Delta}$, and the vector with the number of sites with at least two reads $\mathbf{M}$. Our model has two possible hidden states: homozygous state ($Y_w=4$), and non-homozygous state ($Y_w=2$). As above, we collect the hidden states in a vector $\mathbf{Y} = (Y_1, \dots Y_w, \dots, Y_L)$. The complete data likelihood for the model in this case is then:

\begin{align}\label{eq:10}
    log P(\mathbf{\Delta},\mathbf{Y}|\Theta,\bm{\delta}) &= log P(\mathbf{\Delta}|\mathbf{Y},\Theta, \bm{\delta}) + log P(\mathbf{Y}|\Theta)\nonumber\\
 &= [\sum_{w} log P(\Delta_w|Y_w, M_w, \bm{\delta}) + \sum_{w} log P(Y_w|Y_{w-1}, \bm{A})] + P(Y_0| \bm{\pi}),
\end{align}

where $\Theta$ is a vector of initial probabilities ($\bm{\pi}$), transition matrix ($\bm{A}$) and $\bm{M}$.

Since the source of ROH may not be known, we estimate both transitions and emissions. We calculate the emissions using a beta-binomial likelihood, and fix the mean of the distributions corresponding to $Y_w$ = 4 and $Y_w$ = 2 at expected proportion of differences in a homozygous tract ($p_4$) and expected proportion of differences in a non-homozygous tract ($p_2$) respectively. The expectation step outputs the posterior probability $\Gamma$ of being in state $Y_w = 4$ or the state $Y_w = 2$ in each window. The maximization step for emissions is analogous to that in the KIN model (eq. \ref{eq:4}), and the optimization step here is done with the following cost function:

\begin{align}\label{eq:11}
\mathcal{C} = \sum_{w=1}^L \sum_{i} P(\Delta_w|\delta_{i},p_{i}) \Gamma_{wi} ,
\end{align}
where $P(\Delta_w|\delta_{i},p_{i})$ is a beta-binomial probability with mean $p$ and over-dispersion parameter $\delta$ similar to eq. \ref{eq:3}, and $i$ can take values 2 and 4 corresponding to the hidden states $Y_w$.

To estimate transitions, we initialize the transition matrix with the value 0.2 for the off-diagonal entries, and update it using the standard Baum-Welch update step \cite{baum_maximization_1970}. Similar to the KIN-HMM, we avoid fitting issues by forcing all windows whose proportion of differences is larger than $p_2$ to be in the non-homozygous state (Fig. S7).

\subsection{Contamination correction}\label{contam}
Contamination by DNA from present-day people is a common feature of human ancient DNA datasets \cite{peyregne_present-day_2020}. To address this issue, we developed a heuristic that adjusts both $D_w$ and $N_w$ to minimize the influence of contamination on the relatedness inference.

We assume that contamination rates in both individuals are known and small ($<5\%$), and set $C_{ij} = C_i + C_j$, where $C_i, C_j$ are the contamination estimates from the two individuals. We also assume the divergence $\phi$ between our
target population and the putative contaminant popoulation is known. With probability $C_{ij}$, a comparison between two random reads from the pair of tested individual will contain a contaminant read, and thus contain a difference with probability $\phi$, and with probability $1 - C_{ij}$ it will be between endogenous ones. The comparisons between two contaminant reads are ignored, since we assume $C_{ij}$ to be small.

We  estimate the expected number of differences from comparison of endogenous reads $\mathbb{E}[D_w^{'}]$, and the total number of sites with overlapping endogenous reads $\mathbb{E}[N_w^{'}]$.

For any particular comparison showing a difference $D$ (not to be confused with the number of differences $D_w$), we calculate the probability of the event $E$ that it is between endogenous reads as
\begin{equation}
    P(E|D) = \frac{P(D|E)P(E)}{P(D|E)P(E) + P(D| \neg E)P( \neg E)}.
\end{equation}

Then, by linearity of expectation, we obtain our estimator for the expected number of endogenous comparisons with a difference as
\begin{align}
    \mathbb{E}[D_w^{'}] &= D_{w} P(E|D) = D_{w} \times \frac{P(D|E)P(E)}{P(D|E)P(E) + P(D| \neg E)P( \neg E)}
\end{align}.

Of these terms, $P(E)= 1 -C_{ij} = 1 - P(\neg E)$, and $P(D| \neg E) = \phi$.

For $P(D|E)$, we use an estimator based on the genome-wide average:

\begin{align}
    P(D|E) = \rho = \frac{\frac{\sum_w D_w}{\sum_w N_w} - C_{ij} \phi}{1 - C_{ij}}.
\end{align}

Taken together,

\begin{equation}
 \mathbb{E}[D_w^{'}]= D_{w}\frac{\rho (1-C_{ij})}{\rho(1-C_{ij}) + C_{ij}\phi}
 \end{equation}

 Analogous considerations lead to the expected number of endogenous comparisons that yield no difference:

 \begin{equation}
\mathbb{E}[S_w^{'}] = S_{w} P(E| \neg D) = S_{w} \frac{(1-\rho)(1-C_{ij})}{(1-\rho)(1-C_{ij}) + C_{ij}(1-\phi)}.
 \end{equation}

Here, $S_w = N_w - D_w$. Hence, we set $\mathbb{E}[N_w^{'}] = \mathbb{E}[S_w^{'}] + \mathbb{E}[D_w^{'}]$. We do a similar contamination correction for the input of ROH-HMM.




\section{Results}\label{results}

\subsection{Evaluation with simulations}
We first tested the performance of KIN with simulated pedigrees. We performed coalescent simulations to generate 8 unrelated diploid genomes and artificially mated them to form pedigrees of 17 individuals with relationships up to $5^{th}$ degree (Fig. S3). To evaluate the effect of sequence coverage on the performance of KIN, we generated artificial pileups at each polymorphic site for each individual following a Poisson distribution with 6 different average depths varying between 4$x$ and 0.03$x$ (see section \ref{method}). To mimic ROH, for some pedigrees, we picked a single allele at heterozygous sites in some regions as determined by a Markov chain so that on an average about $17\%$ of the genome is ROH. We also created versions with contamination, by introducing alleles from distantly related individuals, and ascertainment bias, by selecting polymorphic sites identified in a subset of the individuals (see section \ref{method}). We created 60 pedigrees for each combination of average coverage and scenarios of presence/absence of ROH, contamination, and ascertainment bias, totalling 2,880 pedigrees.

\paragraph{ROH detection:}
In Fig. \ref{fig2:ROH}, we present an example of data and inference of the ROH-HMM for simulations with and without ROH, potentially with ascertainment bias and contamination. In all cases, we find that the inferred ROH closely matches the simulations, but the confidence in the classification tends to increase with the simulated coverage.

A systematic evaluation of the performance of the ROH-HMM is given in Table \ref{tab:Table 2}: For the purpose of this analysis, we classified all windows that were at least 20\% homozygous as ROH, and the remainder as non-ROH. Likewise, we classified all windows with a posterior probability of ROH of at least 20\% as ROH. We used these cases to compute sensitivity (Se) and specificity (Sp). In the control case (simulations with no ascertainment, contamination, or ROH) we see that the specificity remains $\geq0.99$ for all coverages. In the cases where we simulate ROH, we find that sensitivity decreases from 97\% at 4x coverage to less than 65\% at 0.05x coverage, but specificity remains high at above 0.96 for all cases, suggesting that the number of erroneously called ROH segments is low.

\begin{figure}[ht!]
    \includegraphics[width=16.5cm]{plots/inkscape_finalImg/ROHplot_sub.png}
    \centering
    \caption{Estimation of ROH probabilities along the genome in simulations. The top row in each panel shows the proportion of differences in a simulated individual along the genome. In the bottom row, the dotted line shows the proportion of each window in ROH, and the solid lines show the estimated probability of not being in the ROH state. (A) Simulation with no ascertainment, contamination, or ROH. (B) Simulation with ROH. (C) Simulation with ROH and ascertainment. (D) Simulation with ROH and contamination.}
    \label{fig2:ROH}
\end{figure}

\begin{table}[ht!]
\caption{\label{tab:Table 2}Model performance for ROH prediction. Here we test ROH-HMM in four different cases of simulations: Control (without ROH, ascertainment bias or contamination), R (with ROH), RA (with ROH and ascertainment bias), RC (with ROH and contamination)}
\begin{tabular}{|c|c|c|c|c|c|c|c|}
    \hline
    Coverage & Control (Sp) & R (Se) & R (Sp) & RA (Se) & RA (Sp) & RC (Se) & RC (Sp)\\
    \hline
    4x & $>$0.99 & 0.97 & 0.98 & 0.97 & 0.98 & 0.97 & 0.97\\
    \hline
    0.5x & $>$0.99 & 0.96 & 0.98 & 0.96 & 0.98 & 0.96 & 0.97\\
    \hline
    0.2x & $>$0.99 & 0.92 & 0.98 & 0.91 & 0.98 & 0.93 & 0.97\\
    \hline
    0.1x & $>$0.99 & 0.85 & 0.98 & 0.83 & 0.97 & 0.82 & 0.97\\
    \hline
    0.05x & 0.99 & 0.65 & 0.96 & 0.55 & 0.96 & 0.57 & 0.96\\
    \hline
\end{tabular}
\label{table2}
\end{table}


\paragraph{IBD prediction:}
We investigated the accuracy of IBD state prediction along the genome by counting the number of genomic windows where we correctly predict IBD state for different relatedness and coverages (Fig. \ref{fig4:IBDstate_accuracy}). The accuracy for coverages of 0.1x or higher is consistent and varies between 1.0 and 0.78, depending on relatedness. However, the accuracy decreases at lower coverages for most relatedness cases and ranges from 1.0 to 0.48 at 0.03x. The exceptions are for identical individuals and parent-child pairs, where the accuracy is nearly perfect for all investigated coverages, even at 0.03x coverage. We note that the accuracy of IBD prediction is lowest for siblings, followed by 2nd degree and 3rd degree relatives. This is because IBD in siblings is more variable, and therefore harder to predict. A pair of 3rd-degree relatives, for example, are expected to only share $12.5\%$ of their genome IBD, with the rest being unrelated. Therefore, even a naive classifier that classified everything as unrelated would have an accuracy of $87.5\%$. In contrast, siblings are expected to share one chromosome IBD for half their genomes, and both chromosomes for another $25\%$. This higher variability makes predictions harder.

We find that adding contamination, ascertainment bias in our simulations has little effect on IBD prediction. Adding ROH to the simulations reduces the IBD prediction in two cases: the average accuracy for second degree decreases from 0.89 to 0.85 and for siblings from 0.81 to 0.70 (Fig. S9). We see this adverse effect of ROH in case of siblings, perhaps because only in case of siblings we do not introduce long ROH directly, but long and short ROH may be present through parents (see section \ref{method}). Unidentified short ROH may cause a difficulty in differentiating between different combinations of IBD ($Z_w$) and ROH states ($H_w$). However, this does not affect the power to identify siblings in presence of ROH even at $0.05x$ (see Fig. \ref{fig3:Comparison_READ_KIN})


\begin{figure}[ht!]
    \includegraphics[width=16.5cm]{plots/plotimg/plot_IBDaccuracy.png}
    \centering
    \caption{Evaluation of IBD estimation at different coverages. y-axis shows accuracy calculated over 60 simulations with each relatedness case and six different coverages shown on x-axis. The error bars are drawn at 1 standard deviation from the mean. Here, accuracy corresponding to relatedness cases for Parent-Child and Identical individuals is always 1, and overlaps with each other. Accuracy is defined as the proportion of correctly predicted IBD states, when compared to the central position of the window.}
    \label{fig4:IBDstate_accuracy}
\end{figure}




\paragraph{Relatedness classification:}
We evaluated the classification accuracy of KIN (cutoff: 1 log-likelihood unit) and compared it to that of READ (cutoff: 1 standard deviation)) (Fig. \ref{fig3:Comparison_READ_KIN}). We first describe the results for relatedness cases detectable by READ, viz. identical, $1^{st}$ degree, $2^{nd}$ degree and unrelated. In this case we show that both methods have similar performance for low-coverage shotgun data (``control''-case), and for ascertained data. The true positive rate is above 0.97 for both KIN and READ, while the false negative rate is below 0.02. One exception is  $2^{nd}$-degree relatives, where KIN has higher power than READ, and as coverage decreases from 4x to 0.05x, KIN's true positive rate decreases from 0.95 to 0.89, compared to range of 0.91 to 0.75 for READ. The false positive rate in this case remains below 0.03 for both methods.

To investigate the impact of contamination, we performed simulations where we added up to 3\% contamination to some individuals. We also find that READ is strongly impacted by contamination  as the true positive rate is in the range 0.89 to 0.78 and 0.54 to 0.46 for 1st and 2nd degree relatives, respectively, and the false positive rate reaches up to 0.04 and 0.17 for unrelated individuals and 2nd degree relatives, respectively. In comparison, we also ran KIN, giving it the simulated contamination amounts for each individual. We found that the correction implemented in KIN is sufficient to remove this bias, and the true and false positive rates remain as in the control (Fig. \ref{fig3:Comparison_READ_KIN}). We also perform an analysis where we misspecify the contamination, and find that the effect of modest misspecification is small (Fig. S13).

For simulations with ROH, we find that KIN also outperforms READ for 1st and 2nd degree relatives, although both the true and false positive rates increased for both methods compared to the control for 2nd degree relatives. The increase in false positives is likely due to ROH making related individuals more similar. Finally, we also found that KIN has good power to detect relatedness cases that are not detectable by READ, i.e. parent-child, siblings and, to a lower level, 3rd degree relatives (Fig. \ref{fig3:Comparison_READ_KIN}).

\begin{figure}[ht!]
    \includegraphics[width=16.5cm]{plots/plotimg/comparison_plot.png}
    \centering
    \caption{Comparison of KIN with READ using simulations with different coverages, and different cases of ascertainment, contamination and ROH. Unrelated label here refers to KIN performance results when all Unrelated, Fifth Degree, Fourth Degree, Third Degree pairs are labelled as Unrelated (for fair comparison with READ). 'Unrelated w/o $3^{rd}$ Degree' refers to the performance results when $3^{rd}$ Degree is classified separately from the unrelated individuals.}
    \label{fig3:Comparison_READ_KIN}
\end{figure}

\subsection{Application to real data}

\paragraph{Chagyrskaya and Okladnikov Neandertals:}
To test KIN on real ancient data, we applied it to a Neandertal dataset from Chagyrskaya and Okladnikov Caves in Siberia, Russia \cite{kolobova_archaeological_2020-1, mafessoni_high-coverage_2020,skov_genetic_2022}. This dataset contains genetic data from a total of 16 skeletal remains that likely belong to contemporary Neandertals who occupied the Chagyrskaya and Okladnikov caves between 59 and 51 kya, and at least 44 kya respectively \cite{skov_genetic_2022}. DNA extracted from each of these remains were captured with an array targeting variable sites identified in high coverage Neandertal and Denisovan genomes, and common variations in Africans \cite{skov_genetic_2022}. This genetic data has  low-to-intermediate depth of coverage ranging from 0.01x to 12.34x, with 8 samples at  $<1x$ coverage. Some of these specimens showed signs of long ROH, and DNA contamination from modern humans as well as hyenas \cite{skov_genetic_2022}. We focused our analysis on the variable sites in two high-coverage Neandertal genomes: Altai Neandertal (Denisova 5) \cite{prufer_complete_2014} and Vindija 33.19 \cite{prufer_high-coverage_2017} as done by the authors for the relatedness analysis \cite{skov_genetic_2022}. Our results for pairwise relatedness for these individuals are shown in Fig.\ref{fig5:Chagyrskaya_KIN}. We found three specimens from the same  individual (Chagyrskaya13-Chagyrskaya19-Chagyrskaya1141), a parent-child pair (Chagyrskaya07 and Chagyrskaya17), and a pair of  $2^{nd}$-degree relatives (Chagyrskaya01/Chagyrskaya60). Further, we identify Chagyrskaya17 and Chagyrskaya60 as 3rd-degree relatives (table S1).

Our estimates are consistent with those obtained using READ, except that READ is unable to detect 3rd-degree relationships, and does not distinguish between sibling and parent-child relationships. We also find that both KIN and READ classify Chagyrskaya06 and Chagyrskaya14 as parent-child with low confidence, but from the morphology they likely stem from the same individual. We believe that the low confidence mis-classification may be due to uncorrected non-human contamination present in these libraries ($1\%$ and $2.9\%$ respectively \cite{skov_genetic_2022}), biasing the estimated differences between the individuals to higher values.

We compared the IBD estimates obtained using KIN to those from lcMLkin for all pairs for whom READ results (with s.d. $>$1) and KIN results ($\Delta LL>$1) match (Fig. S10). We find that lcMLkin creates four different clusters based on the coefficient of relatedness ($r$) and the proportion of the genome in the unrelated state ($k_0)$ for different relatedness cases (identical, parent-child, $2^{nd}$ degree and unrelated). However, the location of these clusters strongly deviate from the expected values, likely due to the low coverage and ROH. Remains from the same individual, for example, are expected to be at $k_0 = 0$ and $r = 1$, but are at $k_0 \approx 0.3$, $r \approx 0.6$ for lcMLkin (Fig. S10).


\begin{figure}[ht!]
    \includegraphics[width=16.5cm]{plots/inkscape_finalImg/kinplot_sub.png}
    \centering
    \caption{Application of KIN to Neandertal remains from Chagyrskaya and Okladnikov Caves. The color of a square represents the relatedness, while the number within denotes log likelihood ratio ($\Delta LL$) between the two maximum likelihood models.}
    \label{fig5:Chagyrskaya_KIN}
\end{figure}

\subsubsection{Ancient modern humans}

We further applied KIN to a genome-wide dataset of 118 ancient individuals from the Lech Valley \cite{mittnik_kinship-based_2019}. We compared our relatedness estimates to those obtained by the authors using READ and lcMLkin. We found that KIN was able to confidently classify $85\%$ of the 6903 possible comparisons ($\Delta LL>1$), READ $94\%$ ($s.d.>1$) and lcMLkin 53\% of pairs (\cite{mittnik_kinship-based_2019}).


Only for 28 pairs, KIN has confident classifications that differ from those obtained using READ (table S2, Fig. S11). Twenty of these comparisons, KIN predicts to be third-degree relationships, and  READ concordantly classifies them as unrelated (as it only infers first and second degree relationships). lcMLkin predicts $3^{rd}$ - $5^{th}$ in 14 of these cases, unrelated in 1 case (see Fig. S11), second degree in 1 case (see Fig. S11), and does not have enough data for classification in 4 cases. For 10 additional pairs, lcMLkin predicts $3^{rd}$ - $5^{th}$ degree, but KIN infers them to be unrelated. There are a total of 83 pairs where READ obtains a confident call ($s.d.>1$), but KIN does not ($\Delta LL<1$). In 80 of these cases, READ classifies them as unrelated, KIN classifies $3^{rd}$ degree, while lcMLkin predicts $3^{rd}$ - $5^{th}$ degree. The remaining cases READ classifies unrelated, while KIN and lcMLkin call a $2^{nd}$-degree relationship. Thus, the vast majority of differences can be explained by READ not considering 3rd-degree relationships.

For less than third-degree relations, only eight cases are classified differently between KIN and READ. lcMLkin matches KIN's prediction in two cases, and does not have enough data in 5 cases. In the last case, all three methods differ (table S3, Fig. S11), and the true relatedness is unresolved in this case. Finally, there are three disagreements between KIN and lcMLkin in classification of parent-child versus siblings (READ predicts first degree for all three pairs). We plotted the pairwise differences for these three pairs in Fig. S12, and found that the proportion of differences along the genome aligns with the prediction of KIN.



\section{Discussion}\label{discussion}
Here, we present a new method called KIN to estimate genetic kinship and the location of IBD tracts from low-coverage data, in presence of long ROH, contamination, as well as ascertainment bias. Our method utilizes a set of HMMs to estimate IBD tracts and uses them to classify each pair of individuals into a possible relatedness case, along with a measure of classification confidence (Fig. S1). We evaluated the method performance of KIN, and compared it to that of READ using simulated pedigrees. Finally, we show applications of KIN on two ancient datasets.

For detecting ROH, there is only one method available that works with low coverage ancient DNA \cite{ringbauer_parental_2021}. This software can infer ROH at coverages $\geq0.3x$, but requires a large reference panel. Instead, our method detects long ROH based on just the expected heterozygosity $p_0$, which can be estimated from a small number of unrelated individuals.  On simulations, we find that our method reliably inferrs ROH regions  with lengths on the order of 10cM from samples sequenced to coverage $\geq 0.1x$ .
In our simulations, we find that adding long ROH ($\approx17\%$) to simulations slightly improves the power of both KIN and READ, particularly at lower coverages. This is likely because ROH actually reduces the variance in differences depending on the relatedness, and makes it easier to correctly classify relatedness. For READ, the presence of ROH causes a bias towards inferring closer relatedness cases, but the model we use in KIN reduces this bias (see Fig. \ref{fig3:Comparison_READ_KIN}).

We show that KIN reliably detects IBD tracts for $\geq0.1x$ coverage even in the presence of contamination and ascertainment bias, and the accuracy for IBD detection reduces with lower coverages. Adding ROH to the simulations adversely affects IBD prediction in siblings (Fig. S9), but this does not affect the power to correctly classify siblings (Fig. \ref{fig3:Comparison_READ_KIN}).

Contamination can affect the accuracy of relatedness inference for two reasons. For one, if a substantial fraction of samples is contaminated, then the estimation of $p_0$ becomes inaccurate, because the majority of pairwise differences will include at least one contaminated sample. The second issue is that contaminated samples will look less similar to other individuals, and thus cause a bias towards inferring them to be less related to other individuals. The amount of contamination does not need to be large for this to be important, in our simulations we find that even at contamination levels $\leq 3\%$, the performance of READ is substantially reduced. When contamination rates are correctly inferred, the correction we implemented leads to improved performance compared to naive methods such as READ, although they too could be ameliorated in a similar way \cite{skov_genetic_2022}. However, in many cases contamination estimates may be uncertain or inaccurate, and we show in fig. S13 that KIN's performance is robust to small deviations (small compared to average pairwise heterozygosity) in contamination estimates.

The Lech Valley data has low contamination, and no ROH. For pairwise comparisons with large numbers of overlapping sites ($>10000$), KIN, READ and lcMLkin all mostly agree. However, KIN is able to differentiate between parent-child and siblings, and identify second degree relationship from just a few thousand polymorphic sites ($\approx$ 4000) overlapping between samples. KIN can also infer third degree relation with $\approx$ 30,000 overlapping polymorphic sites. We show that when applied to Neandertal specimens from Chagyrskaya and Okladnikov Caves, KIN identifies a pair of $1^{st}$-Degree relatives as parent-child, which is in agreement with the finding that the mtDNA haplotypes differ between the samples (one sample is male and the other female) \cite{skov_genetic_2022}. In addition, KIN identifies a pair of $3^{rd}$ degree relatives. In this case of a population with large amounts of ROH, we find that the inference by lcMLkin are heavily biased, but KIN's model takes ROH into account and both the coefficient of relatedness and $k_0$ are very close to what would be expected from the inference by both READ and KIN.

One limitation of our approach is that it assumes a single population. In case of a highly structured population, KIN may show inaccurate inference of $p_0$ causing inaccurate relatedness inference. Also, our method makes the assumption, that the median pairwise genetic difference in the population reflects the population diversity $p_0$, which fails if almost all individuals in the dataset are related. We may get around this problem by using an estimate of $p_0$, calculated from known a pair of unrelated individuals from same population, or another population with similar diversity. We provide the user with an option to give an estimate of $p_0$. The current implementation of KIN is restricted to six relatedness cases we expect to be most common, but it might be feasible to extend it to other cases, such as double first cousins, using a corresponding IBD state transition matrix.

While we have focused on the application of KIN on ancient human samples, the model is not tied to this system. Assuming we know the recombination rate, and hence can estimate the transition matrix (see section \ref{method}), KIN can be widely applied to any diploid species. In addition, the output of KIN is a table which shows for each pair, the most likely model, and the second best guess, along with a confidence level represented by the log likelihood ratio. This makes KIN easy to automatize for large datasets. To make application of KIN user-friendly, we provide a python package (KINgaroo) to create input files for KIN from processed bam files, while optionally estimating ROH, and correcting for contamination estimates.


\section{Materials and methods}\label{method}

\subsection{Simulations}\label{simulat}
We use simulations both for estimating the transition matrices and for testing and validating our algorithm. All simulations are performed in a scenario mimicking the analysis of a Neandertal population contaminated by modern humans \cite{mafessoni_high-coverage_2020}. We simulate unrelated individuals using $\texttt{msprime}$ \cite{kelleher_efficient_2016}, followed by an additional step where we simulate related individuals using a predetermined pedigree (Fig. S3).

\paragraph{Simulating pedigrees:}

For our simulations of background diversity, we form a population (Pop1) with constant effective size of 10,000 and sample eight diploid individuals (each made up of two haploid individuals) from 2500 generations ago (Fig. S3). For each individual, we simulate 22 chromosomes with length $L\approx96$ Mb (same as chromosome 13) and a recombination rate of $r=10^{-8}$ per base pair per generation. We introduce mutations using an infinite sites model with rate  $\mu= 10^{-8}$ per base pair per generation.

For the pedigree simulations, we first simulate a recombined set of chromosomes for either parent, and combine them to create the progeny. There are two different ways in which we generate recombination points. For the estimation of transition probabilities, we simulate recombination by first drawing the number of breakpoints as a Poisson random variable with parameter $rL$, and use a uniform distribution on $[1, L]$ to sample the positions of recombination points. For the testing of our method, we use Ped-sim \cite{caballero_crossover_2019} to simulate recombination points. This allows us to take into account sex-specific recombination rates and crossover interference, and thus is expected to give a more realistic recombination landscape. The pedigree simulations result in nine addition individuals, resulting in a total sample of 17 individuals.

\paragraph{Transition matrices:}
KIN requires a transition matrix for each relatedness case, which we estimate by counting the transition between IBD states for all pairs of individuals with that relatedness in a training set of 1000 simulations from our pedigree (Fig. S3). For two cases, siblings and grandparent-grandchild, it is possible to write down the theoretical expectation of the transition matrix:

For grandparent-grandchild, rate matrix is
 $$Q = \left[\begin{array}
{rrr}
-r & r \\
r & -r
\end{array}\right]$$

Similarly, for a pair of siblings, we calculate $$Q =
\left[\begin{array}
{rrr}
-4r & 4r & 0\\
2r & -4r & 2r\\
0 & 4r & -4r
\end{array}\right]$$

The state space includes all IBD states in this case. For these two cases, we get the transition matrix in each case as $e^{Qb}$, where b is the window size.


\paragraph{Simulations for method evaluation:}
Apart from the related and unrelated individuals in Pop1, we simulated more haploid individuals in three other populations to create scenarios with ascertainment bias and contamination (Fig. S3). We simulated two individuals to form an individual each from two other populations (Pop2, Pop3) with split time of 3500 and 4500 generations with Pop1, and sampling time of 2000 generations and 4000 generations ago respectively.  We identified the sites that were polymorphic among these two individuals, and used these sites to ascertain the genomes of individuals from Pop1. This scenario roughly models the ascertainment of the Chagyrskaya and Okladnikov Caves data. We tested the performance of our method in presence of long ROH ($\sim17\%$), by simulating regions of homozygosity in unrelated individuals with a Markov chain using the transition matrix:

$$\mathbf{A} = \left[\begin{array}
{rr}
1-10/L & 10/L \\
2/L & 1-2/L  \\
\end{array}\right]
$$

It is worth noting that we introduce long ROH in unrelated individuals, before artificially mating them to form pedigrees, which means that we do not directly introduce ROH in progeny, but it still affects relatedness inference among progeny as shown in Fig. \ref{fig0:schematic}.
From the steps described above, we got genotypes of individuals in Pop1 in presence/absence of ROH and ascertainment. We further simulated five diploid individuals from Pop4 with split time of 20,000 generations with Pop1, sampled from the present time, as a source of contamination.  For cases including contamination, we contaminated eight individuals with varying amounts between 0.5\% to 3\% contamination, while the remaining  nine individuals did not have any contamination. We generated reads (derived/ancestral) for different genomic coverages ranging from 4x to 0.03x, assuming a Poisson distribution.

For testing, we replicate our simulation 60 times, and create data from the same base simulations at varying levels of coverages and different scenarios: We have a control scenario (without ascertainment bias, ROH or contamination), and individual scenarios where we add ROH, SNP ascertainment and contamination. For the evaluation of the ROH-HMM, we also combine ROH with ascertainment or contamination.


\section{Acknowledgements}
We thank Svante Pääbo, Janet Kelso, Harald Ringbauer, Johann Visagie, Zbigniew \linebreak Jędrzejewski-Szmek, Laurits Skov, Leonardo N. M. Iasi, Alba Bossoms Mesa, Arev P. Sümer, Zuzana Hofmanová, Guido Alberto Gnecchi Ruscone, Ke Wang and Luca Traverso for helpful comments and discussions. This work was supported by the Max Planck Society and the European Research Council (Grant No.: 694707) to Svante Pääbo.

\section{Contributions}
Conceptualization (Design of study): B.M.P.; Software: D.P.; Methodology—lead: D.P.; Methodology—support: B.M.P., S.P.; Formal Analysis: D.P.; Visualization-lead: D.P.; Visualization-support: S.P.; Data Curation: D.P.; Writing—lead: D.P.; Writing—support: B.M.P., S.P.; Supervision: B.M.P.

\section{Data and material availability}
An open-source implementation of KIN and KINgaroo in python along with a toy example dataset, and the scripts to generate our simulations are available on GitHub \cite{divyaratan_popli_httpsgithubcomdivyaratanpoplikinship_inferencereleasestagv313_2022}. We have deposited the version of the software used in the manuscript, along with the above mentioned files on Zenodo \cite{divyaratan_popli_httpsdoiorg105281zenodo7067142_nodate}. The analysed datasets were generated in previous studies, and are available in European Nucleotide Archive \cite{mittnik_kinship-based_nodate,skov_genetic_nodate}.

\section{Ethics declarations}
\subsection{Ethics approval and consent to participate}
Not applicable.

\subsection{Consent for publication}
Not applicable.

\subsection{Competing interests}
The authors declare that they have no competing interests.


\bibliographystyle{plain}
\bibliography{references.bib}

\end{document}
